openfl/transport/grpc/director_client.py

```diff
@@ -95,6 +95,21 @@ def get_experiment_data(self, experiment_name):
 
         return data_stream
 
+    def set_experiment_failed(
```

Comment:
should user call this RPC?

Reply:
It's the internal director client used by the Envoy. User shouldn't use this RPC. Probably stop experiment, pause, delete, but not set failed.

Reply:
is user notified somehow that experiment failed?

Reply:
No, we haven't a notification mechanism yet. But user can indirectly understand it by stopping StreamMetric and we write the corresponding error message to the director log.
The purpose of this PR is to avoid federation freezes after collaborator failures, not to implement a complete monitor system.